Framework/Kernel/inc/MantidKernel/DataService.h

```diff
@@ -19,6 +19,15 @@
 
 namespace Mantid {
 namespace Kernel {
+
+/// Flag for whether to sort items before returning
+enum class DataServiceSort { Sorted, Unsorted };
+/**
+ * Flag for whether to include hidden items when returning,
+ * Auto queries the class to determine this behavior
+ */
+enum class DataServiceHidden { Auto, Include, Exclude };
```

Comment:
Ah, I hadn't considered that here there's the additional complexity of the template. That makes sense as a good compromise.